.github/workflows/release-sdk.yml

```diff
@@ -3,15 +3,66 @@ name: Build and Release W&B SDK
 on:
   workflow_dispatch:
     inputs:
-      use_pre:
+      version:
+        type: string
+        description: "version to assign to the release"
+        required: true
+      update-changelog:
         type: boolean
-        description: "pre-release"
+        description: "update the changelog and release notes"
         required: false
         default: false
 
 jobs:
+  prepare-release:
+    name: Prepare Release
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+
+      - name: Setup git config
+        run: |
+          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git remote set-url origin https://${{ secrets.GITHUB_TOKEN }}@github.com/wandb/wandb.git
+
+      - name: Install bump2version
+        run: pip install bump2version
+
+      - name: Create and Switch to Release Branch
+        run: |
+          git checkout -b release-${{ github.event.inputs.version }}
+
+      - name: Bump Version
+        run: |
+          bump2version patch --no-tag --no-commit --config-file .bumpversion.wandb.cfg --new-version ${{ github.event.inputs.version }}
+
+      - name: Update CHANGELOG.md
+        if: ${{ inputs.update-changelog }}
+        run: |
+          DATE=$(date +'%Y-%m-%d')
+          sed -i "s/## Unreleased/## Unreleased\n\n## [${{ github.event.inputs.version }}] - $DATE/" CHANGELOG.md
```

Comment:
nice! we may want to replace all the way to the end of the "Unreleased" line in case there's any text tacked on at the end

Reply:
yeah, but i hoped that we would just add these things as md comments.
but you are right to be fully safe that would be better.